library/core/src/fmt/mod.rs

```diff
@@ -119,6 +119,13 @@ pub trait Write {
     ///
     /// This function will return an instance of [`Error`] on error.
     ///
+    /// Though it is possible for implementors of this trait to return an error, at the time
+    /// of writing these docs, no implementation of [`std::fmt::Write`] in the standard library
+    /// returns such an error.
```

Comment:
This claim is false. [The internal implementation of `std::fmt::Write` for an `Adapter`](https://github.com/rust-lang/rust/blob/cc4dd6fc9f1a5c798df269933c7e442b79661a86/library/std/src/io/mod.rs#L1665) that wraps a `std::io::Write` produces a `fmt::Error` when the underlying destination returns a `std::io::Error` (while saving that error internally and returning it after the formatting is cancelled).

This can be observed in user code by implementing a formatting trait which is then used with `Write::write_fmt()` on a destination that errors.

I think better advice to give here is something like “The purpose of `std::fmt::Error` is to abort the formatting operation when the underlying destination encounters some error preventing it from accepting more text; it should generally be propagated rather than handled, at least when implementing formatting traits.”

Reply:
Oh, I missed that! Thanks for the heads up. I'll make the change with your suggestion, run the checks, and push the commit.